modules/gdscript/gdscript_parser.cpp

```diff
@@ -4181,6 +4182,33 @@ bool GDScriptParser::export_annotations(const AnnotationNode *p_annotation, Node
 	return true;
 }
 
+bool GDScriptParser::export_custom_annotation(const AnnotationNode *p_annotation, Node *p_node, ClassNode *p_class) {
+	ERR_FAIL_COND_V_MSG(p_node->type != Node::VARIABLE, false, vformat(R"("%s" annotation can only be applied to variables.)", p_annotation->name));
+
+	VariableNode *variable = static_cast<VariableNode *>(p_node);
+	if (variable->exported) {
+		push_error(vformat(R"(Annotation "%s" cannot be used with another "@export" annotation.)", p_annotation->name), p_annotation);
+		return false;
+	}
+	if (p_annotation->resolved_arguments.size() < 2) {
+		push_error(R"(Annotation "@export_custom" requires 2 arguments.)", p_annotation);
+		return false;
+	}
```

Comment:
The arguments are validated before the annotation is applied, so this can be replaced with `ERR_FAIL_COND_V_MSG`.